src/components/modals/export/MtsSysexExport.vue

```diff
@@ -0,0 +1,116 @@
+<script setup lang="ts">
+import MtsSysexExporter from "@/exporters/mts-sysex";
+import { sanitizeFilename } from "@/utils";
+import { ref } from "vue";
+import Modal from "@/components/ModalDialog.vue";
+import type { Scale } from "scale-workshop-core";
+import { clamp } from "xen-dev-utils";
+
+const props = defineProps<{
+  newline: string;
+  scaleName: string;
+  baseMidiNote: number;
+  scale: Scale;
+}>();
+
+const emit = defineEmits(["confirm", "cancel"]);
+
+function clampName(name: string): string {
+  return name.slice(0, 16);
+}
+
+function formatPresetIndex(input: string | number): number {
+  let number =
+    typeof input === "string" ? parseInt(input.replace(/\D/g, "")) : input;
+  if (Number.isNaN(number)) return 0;
+  return clamp(0, 127, number);
+}
+
+// Rarely implemented parameters
+// const deviceId = ref(0);
+// const bank = ref(0); (only for message 0x0804)
+
+const name = ref(clampName(props.scaleName));
+const presetIndex = ref(0);
+
+function doExport() {
+  const params = {
+    newline: props.newline,
+    scale: props.scale,
+    filename: sanitizeFilename(props.scaleName),
+    baseMidiNote: props.baseMidiNote,
+    name: name.value,
+    presetIndex: presetIndex.value,
+  };
+
+  const exporter = new MtsSysexExporter(params);
+  exporter.saveFile();
+
+  emit("confirm");
+}
+</script>
+
+<template>
+  <Modal @confirm="doExport" @cancel="$emit('cancel')">
+    <template #header>
+      <h2>Export MTS Bulk Tuning Dump</h2>
+    </template>
+    <template #body>
+      <div class="control-group">
+        <div class="control">
+          <label for="name">Name (16-character limit)</label>
+          <input
+            class="half"
+            type="text"
+            id="name"
+            v-model="name"
+            maxlength="16"
```

Comment:
Are you sure this is enough validation? I'm unable to break it myself, but you could enforce `clampName` everytime the ref is updated.

Reply:
That must have been an oversight after I realized how to do it for presetIndex. I'm used to using computed properties in Vue2 for this kinda stuff, so I was toying around with it before I figured it out.